### ISSUE TITLE
Reset to first slice button?

### ISSUE THREAD

**Issue by mxmilkiib** [sjaehn/BJumblr#3]
So that if the user finds that their loop is not in sync with the music that is being input, they can press the reset-to-first-beat button and the left-most column gets started from that point?

(A related feature suggestion with similar but different functionality that it might be worth mentioning at the same time: to have -1 or +1 buttons, so as to shift each columns samples a column left or right? That would not change the start time of the samples/loop though.)

**Comment by sjaehn** [comment]
Integer precision is sufficient? I would shift the white cursor line.

**Comment by sjaehn** [comment]
Like this: 2e9d7ce0286d155f9afe322350b563164bb55d85

**Comment by mxmilkiib** [comment]
I don't think the buttons are working, there is no change.

I have had the thought though - how much is this connected to the timebase of the host? Carla has a b/b/t counter. Maybe that should up to be the host to reset the bar start beat? (Can one do that from an LV2 plugin?)

**Comment by sjaehn** [comment]
They are working, but not in the expected way. As the plugin DSP uses the deviation from the default diagonal line to calculate which audio fragment to play, there will be no change if you simply step forward or backward (at least with the default diagonal line). Bugfix will follow.

The plugin always synchronizes with the host time/position (Carla b/b/t) information. Usually some 20 times per second. Depending on system, host, weather outside, ... The new buttons simply add an offset to this synchronized position. In case of the - and the + button, there is simply 1 step added (or diminished). In case of the < button it is minus actual step position. And in case of the home button offset is zero.

**Comment by sjaehn** [comment]
I intensely tested the plugin sync behavior using a counting metronome input ("1"-"2"-"3"-"4"). As result I removed the bug flag as it is not a bug (unless you get different results than I did). The plugin exactly did what I programmed. I think we have different expectations what the plugin sync functions should do.

If I press the "<" button, the white line jumps to step 1 and the audio output now starts from the slice it jumped from. As it should do.

If I press the home button, the line jumps back by the number of steps it has been changed before. Playback in the old order. As I expect.

"+" and "-" shift the line to the right or to the left, respectively. This also means that step 1 starts with one slice before or after, respectively. It needs a bit longer to understand the output. But it is right.

If you press the sync buttons and you use the default diagonal line pattern, there will be no change in audio output. As I mentioned before. Think about why! Only the white line is moved. And thus the starting slice of the pattern. As it should do.

The testing results:
![screenshot](https://user-images.githubusercontent.com/42939779/72935187-2631a380-3d65-11ea-9afb-8c9d436f9d7b.png)
I hear "1"-"2"-"1"-"3"-"1"-"4"-"1"-"1".

"<" button on step 4: jump to step 1 with output "4"-"1"-"4"-"2"-"4"-"3"...
home button: initial output
"+" button on step 4: jump to step 5 with output "4"-"3"-"4"-"4"- jump to position 1 -"4"-"1"...

What did you expect, what should the plugin do?
